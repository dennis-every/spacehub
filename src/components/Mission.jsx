import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../redux/missions/missionsSlice';

const Mission = ({ mission }) => {
  const dispatch = useDispatch();

  const joinMissionHandler = (id) => {
    dispatch(joinMission(id));
  };

  const leaveMissionHandler = (id) => {
    dispatch(leaveMission(id));
  };

  return (
    <tr>
      <td>{mission.mission_name}</td>
      <td>{mission.description}</td>
      <td>
        {mission.reserved && (
          <button
            className="btn btn-info btn-sm"
            style={{ color: 'white' }}
            type="button"
          >
            Active Member
          </button>
        )}
        {!mission.reserved && (
          <button
            className="btn btn-secondary btn-sm"
            type="button"
          >
            Not a Member
          </button>
        )}
      </td>
      <td>
        {mission.reserved && (
          <button
            className="btn btn-outline-danger btn-sm"
            type="button"
            onClick={() => leaveMissionHandler(mission.mission_id)}
          >
            Leave Mission
          </button>
        )}
        {!mission.reserved && (
          <button
            className="btn btn-outline-secondary btn-sm"
            type="button"
            onClick={() => joinMissionHandler(mission.mission_id)}
          >
            Join Mission
          </button>
        )}
      </td>
    </tr>
  );
};

Mission.propTypes = {
  mission: PropTypes.shape({
    mission_id: PropTypes.string.isRequired,
    mission_name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool,
  }),
};

Mission.defaultProps = {
  mission: {
    reserved: false,
  },
};

export default Mission;
