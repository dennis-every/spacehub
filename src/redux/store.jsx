import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketsReducer,
  },
});

export default store;
