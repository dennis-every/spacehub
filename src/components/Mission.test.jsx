import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Mission from './Mission';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

const missionObject = {
  mission_name: 'Thaicom',
  mission_id: '9D1B7E0',
  description: 'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
};

describe('Test suite for the Mission component', () => {
  it('Expect to render a Snapshot of the Mission component', () => {
    const mission = render(
      <Provider store={store}>
        <table>
          <tbody>
            <tr>
              <Mission key={missionObject.mission_id} mission={missionObject} />
            </tr>
          </tbody>
        </table>
      </Provider>,
    );
    expect(mission).toMatchSnapshot();
  });

  it('Expect to test if the text "Join Mission" is inside the Mission component', () => {
    render(
      <Provider store={store}>
        <table>
          <tbody>
            <tr>
              <Mission key={missionObject.mission_id} mission={missionObject} />
            </tr>
          </tbody>
        </table>
      </Provider>,
    );
    expect(screen.getByText('Join Mission')).toBeInTheDocument();
  });
});
