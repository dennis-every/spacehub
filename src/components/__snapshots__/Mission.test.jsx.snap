// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the Mission component Expect to render a Snapshot of the Mission component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table>
        <tbody>
          <tr>
            <td>
              Thaicom
            </td>
            <td>
              Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
            </td>
            <td>
              <button
                class="btn btn-secondary btn-sm"
                type="button"
              >
                Not a Member
              </button>
            </td>
            <td>
              <button
                class="btn btn-outline-secondary btn-sm"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table>
      <tbody>
        <tr>
          <td>
            Thaicom
          </td>
          <td>
            Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
          </td>
          <td>
            <button
              class="btn btn-secondary btn-sm"
              type="button"
            >
              Not a Member
            </button>
          </td>
          <td>
            <button
              class="btn btn-outline-secondary btn-sm"
              type="button"
            >
              Join Mission
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
